pkg/controllers/scheduler/scheduler.go

```diff
@@ -761,29 +772,39 @@ func (s *Scheduler) enqueueFederatedObjectsForCluster(cluster *fedcorev1a1.Feder
 	logger := s.logger.WithValues("cluster", cluster.GetName())
 
 	if !clusterutil.IsClusterJoined(&cluster.Status) {
-		s.logger.WithValues("cluster", cluster.Name).
-			V(3).
-			Info("Skip enqueue federated objects for cluster, cluster not joined")
+		logger.V(3).Info("Skip enqueue federated objects for cluster, cluster not joined")
 		return
 	}
 
 	logger.V(2).Info("Enqueue federated objects for cluster")
 
-	fedObjects, err := s.fedObjectInformer.Lister().List(labels.Everything())
-	if err != nil {
-		s.logger.Error(err, "Failed to enquue FederatedObjects for policy")
-		return
-	}
-	for _, obj := range fedObjects {
-		s.worker.EnqueueObject(obj)
-	}
-	clusterFedObjects, err := s.clusterFedObjectInformer.Lister().List(labels.Everything())
-	if err != nil {
-		s.logger.Error(err, "Failed to enquue ClusterFederatedObjects for policy")
-		return
-	}
-	for _, obj := range clusterFedObjects {
-		s.worker.EnqueueObject(obj)
+	for _, policyLabel := range []string{PropagationPolicyNameLabel, ClusterPropagationPolicyNameLabel} {
+		hasPolicy, err := labels.NewRequirement(policyLabel, selection.Exists, []string{})
+		if err != nil {
+			logger.Error(err, "Failed to generate label selector for federated objects")
+			return
+		}
+		if err != nil {
+			logger.Error(err, "Failed to generate label selector for federated objects")
+		}
+		labelSelector := labels.NewSelector().Add(*hasPolicy)
+
+		fedObjects, err := s.fedObjectInformer.Lister().List(labelSelector)
```

Comment:
```suggestion
		labelSelector := labels.Everything()
		if r, err := labels.NewRequirement(policyLabel, selection.Exists, nil); err == nil {
			labelSelector = labels.NewSelector().Add(*r)
		}

		fedObjects, err := s.fedObjectInformer.Lister().List(labelSelector)
```

Reply:
wait.. it could make everything requeue twice..ignore my suggestion, but you have to remove duplicate err